### PR TITLE
Binary content update:

### PR DESCRIPTION
**Pull request by wagmarcel**
*Updated springframework and dependencies to v2.1.2. Needed for adding default cbor http body decoder
*Added new gradle version to Dockerfile to work with most recent springframework
*Pass in dataType to differentiate between binary and non binary data
*Search request: Added cbor as return value options. Moved return data type from String to generic Objcet to allow Buffers
*Added IllegalDataInquiryArgumentException to DataRetriever
*Added basic data types and validity checks to ComponentDataType class
*Fixed Unit tests to handle DtaTypes
*Cleanup and raising the bar for pmdmain's paremeter number checks to 9

Signed-off-by: Marcel Wagner <wagmarcel@web.de>

### CODE REVIEW

File: src/main/java/com/oisp/databackend/api/DataSubmissionService.java [hunk 1]
```diff
@@ -56,10 +57,11 @@ public Service withParams(String accountId, DataSubmissionRequest request) {
 
     @Override
     public DataSubmissionResponse invoke() throws MissingDataSubmissionArgumentException {
-        logger.error("=============DataSubmissionResponse");
+        logger.debug("=============DataSubmissionResponse");
```
Comment (arkocal) [inline src/main/java/com/oisp/databackend/api/DataSubmissionService.java:60]: I love this change!

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,16 @@
-FROM ubuntu:xenial
+FROM ubuntu:18.04
 RUN apt-get update -qq && apt-get upgrade -y
 
-RUN apt-get install -y curl unzip build-essential openjdk-8-jdk-headless gradle
+RUN apt-get install -y curl unzip build-essential openjdk-8-jdk-headless wget
```
Comment (arkocal) [inline Dockerfile:4]: Do we really need to maintain the builder container? Did you have a look at this: https://hub.docker.com/_/gradle?tab=description  ?

It is fine for the PR, but if there is no concrete reason we could (and probably should) use the standard Gradle image.

Comment (oguzcankirmemis) [reply Dockerfile:4]: I had tried that without it last week, and it did not work straight away, so I believe we need it but I am also not the expert on the topic.

Comment (arkocal) [reply Dockerfile:4]: @oguzcankirmemis  What was the problem?

Comment (oguzcankirmemis) [reply Dockerfile:4]: Some package was simply missing (it was using the cache at this point), but I did not get a look on which it was, the error message was too long, so I had to force docker build without cache.

Comment (arkocal) [reply Dockerfile:4]: This is a known issue. If the update line is cached, the new package lists are not fetched, and therefore packages are not found.
My question is rather about whether we need the (custom) container at all.

Comment (wagmarcel) [reply Dockerfile:4]: I needed a very recent gradle version to work with the most recent Springframework plugin. With the default gradle from ubuntu image it did not build properly. That is why I decided to add the gradle version.

**Review by arkocal — APPROVED** [review]

**Pull request merged**